File: backend/schemas/analytics.py
from pydantic import BaseModel, Field


class CampaignAnalytics(BaseModel):
    campaign_id: int

    total_queue_items: int = 0

    pending: int = 0
    scheduled: int = 0
    calling: int = 0

    connected: int = 0
    completed: int = 0

    no_answer: int = 0
    busy: int = 0
    voicemail: int = 0
    dropped: int = 0

    retry_scheduled: int = 0
    callback_scheduled: int = 0

    escalated: int = 0
    failed: int = 0
    cancelled: int = 0


class RiskDistribution(BaseModel):
    low: int = 0
    medium: int = 0
    high: int = 0
    critical: int = 0


class EscalationAnalytics(BaseModel):
    total: int = 0
    open: int = 0
    in_review: int = 0
    resolved: int = 0
    closed: int = 0

    critical: int = 0
    high: int = 0
    medium: int = 0
    low: int = 0


class HospitalAnalytics(BaseModel):
    total_patients: int = 0
    active_patients: int = 0

    total_campaigns: int = 0
    running_campaigns: int = 0

    total_calls: int = 0
    completed_calls: int = 0

    active_calls: int = 0

    risk_distribution: RiskDistribution

    escalations: EscalationAnalytics

File: frontend/src/api/analytics.ts
import apiClient from "./client"
import { MOCK_HOSPITAL_ANALYTICS } from "../mock/mockData"

export interface HospitalAnalytics {
  total_patients: number
  total_campaigns: number
  total_calls: number
  active_calls: number

  risk_distribution: {
    low: number
    medium: number
    high: number
    critical: number
  }

  escalations?: {
    total: number
    open: number
    in_review: number
    resolved: number
    closed: number
    critical: number
    high: number
    medium: number
    low: number
  }

  total_escalations?: number
}

export interface CampaignAnalytics {
  campaign_id: number
  total_queue_items: number

  status_counts?: {
    [key: string]: number
  }

  risk_distribution: {
    low: number
    medium: number
    high: number
    critical: number
  }

  total_escalations?: number
}

export async function getHospitalAnalytics(): Promise<HospitalAnalytics> {
  try {
    const response = await apiClient.get("/analytics/hospital")
    const data = response.data
    if (data && (data.total_patients > 0 || data.total_calls > 0)) {
      if (data.total_escalations === undefined && data.escalations?.total !== undefined) {
        data.total_escalations = data.escalations.total
      }
      return data
    }
    return MOCK_HOSPITAL_ANALYTICS
  } catch {
    return MOCK_HOSPITAL_ANALYTICS
  }
}

export async function getCampaignAnalytics(
  campaignId: number
): Promise<CampaignAnalytics> {
  try {
    const response = await apiClient.get(
      `/analytics/campaign/${campaignId}`
    )
    if (response.data) return response.data
  } catch {
    // ignore
  }

  return {
    campaign_id: campaignId,
    total_queue_items: 24,
    status_counts: {
      PENDING: 8,
      CALLING: 2,
      CONNECTED: 10,
      ESCALATED: 2,
      COMPLETED: 2,
    },
    risk_distribution: {
      low: 15,
      medium: 6,
      high: 2,
      critical: 1,
    },
    total_escalations: 3,
  }
}
